### PR TITLE
[ticket/12487] Update PHP code

### PR DESCRIPTION
**Pull request by MGaetan89**
https://tracker.phpbb.com/browse/PHPBB3-12487

PHPBB3-12487

### CODE REVIEW

**Comment by MGaetan89** [comment]
Does someone have an idea why the test fails?

```
1) quick_links_test
NoSuchElementException: {"errorMessage":"Unable to find element with id 'main'","request":{"headers":{"Accept":"application/json","Content-Length":"29","Content-Type":"application/json;charset=UTF-8","Host":"127.0.0.1:8910"},"httpVersion":"1.1","method":"POST","post":"{\"using\":\"id\",\"value\":\"main\"}","url":"/element","urlParsed":{"anchor":"","query":"","file":"element","directory":"/","path":"/element","relative":"/element","port":"","host":"","password":"","user":"","userInfo":"","authority":"","protocol":"","source":"/element","queryKey":{},"chunks":["element"]},"urlOriginal":"/session/89c2a3b0-92c3-11e4-8983-9d8f5cc71dc2/element"}}
/home/travis/build/phpbb/phpbb/tests/vendor/facebook/webdriver/lib/WebDriverExceptions.php:56
/home/travis/build/phpbb/phpbb/tests/vendor/facebook/webdriver/lib/remote/HttpCommandExecutor.php:225
/home/travis/build/phpbb/phpbb/tests/vendor/facebook/webdriver/lib/remote/RemoteWebDriver.php:471
/home/travis/build/phpbb/phpbb/tests/vendor/facebook/webdriver/lib/remote/RemoteWebDriver.php:127
/home/travis/build/phpbb/phpbb/tests/test_framework/phpbb_ui_test_case.php:90
/home/travis/build/phpbb/phpbb/tests/test_framework/phpbb_ui_test_case.php:200
/home/travis/build/phpbb/phpbb/tests/test_framework/phpbb_ui_test_case.php:72
```

**Comment by MGaetan89** [comment]
This PR needs to be milestoned for 3.2.

**Comment by nickvergessen** [comment]
I'd prefer if you make 1 PR per thing you change, or one PR per file group (e.g. includes/acp, includes, phpbb/, etc)

**Comment by MGaetan89** [comment]
I will make a PR for each change. Can I keep the same ticket for each PR, or should I use multiple tickets?

**Comment by nickvergessen** [comment]
One ticket per issue would be prefered

**Comment by MGaetan89** [comment]
I created tickets [PHPBB3-13454](https://tracker.phpbb.com/browse/PHPBB3-13454) (unused variables, https://github.com/phpbb/phpbb/pull/3255) and [PHPBB3-13455](https://tracker.phpbb.com/browse/PHPBB3-13455) (phpBB deprecated calls).
I keep this PR to update PHP code for PHP 5.3.3+.

File: phpBB/includes/functions.php [hunk 1]
```diff
@@ -3648,26 +3647,19 @@ function phpbb_checkdnsrr($host, $type = 'MX')
 	// checkdnsrr() supports AAAA since 5.0.0
 	// checkdnsrr() supports TXT since 5.2.4
 	if (
-		($type == 'MX' || DIRECTORY_SEPARATOR != '\\' || version_compare(PHP_VERSION, '5.3.3', '>=')) &&
-		($type != 'AAAA' || version_compare(PHP_VERSION, '5.0.0', '>=')) &&
-		($type != 'TXT' || version_compare(PHP_VERSION, '5.2.4', '>=')) &&
-		function_exists('checkdnsrr')
+		($type == 'MX' || DIRECTORY_SEPARATOR != '\\') &&
+		$type != 'AAAA' && $type != 'TXT' && function_exists('checkdnsrr')
```
Comment (nickvergessen) [inline phpBB/includes/functions.php]: this looks incorrect to me ;)

File: phpBB/includes/functions.php [hunk 2]
```diff
@@ -3648,26 +3647,19 @@ function phpbb_checkdnsrr($host, $type = 'MX')
...
 	// Call dns_get_record() if
 	// we're not looking for an AAAA record or
-	// we're not on Windows or
-	// we're running a PHP version where AAAA lookups work reliable
-	if (
-		($type != 'AAAA' || DIRECTORY_SEPARATOR != '\\' || version_compare(PHP_VERSION, '5.3.1', '>=')) &&
-		function_exists('dns_get_record')
-	)
+	// we're not on Windows
+	if (($type != 'AAAA' || DIRECTORY_SEPARATOR != '\\') && function_exists('dns_get_record'))
```
Comment (nickvergessen) [inline phpBB/includes/functions.php]: same here

File: phpBB/includes/functions_jabber.php [hunk 3]
```diff
@@ -87,8 +87,7 @@ function jabber($server, $port, $username, $password, $use_ssl = false)
 	*/
 	static public function can_use_ssl()
 	{
-		// Will not work with PHP >= 5.2.1 or < 5.2.3RC2 until timeout problem with ssl hasn't been fixed (http://bugs.php.net/41236)
-		return ((version_compare(PHP_VERSION, '5.2.1', '<') || version_compare(PHP_VERSION, '5.2.3RC2', '>=')) && @extension_loaded('openssl')) ? true : false;
+		return @extension_loaded('openssl') ? true : false;
```
Comment (nickvergessen) [inline phpBB/includes/functions_jabber.php]: `return (bool) @extension_loaded('openssl');` ?

File: phpBB/includes/functions_messenger.php [hunk 4]
```diff
@@ -44,10 +42,6 @@ function messenger($use_queue = true)
 
 		$this->use_queue = (!$config['email_package_size']) ? false : $use_queue;
 		$this->subject = '';
-
-		// Determine EOL character (\n for UNIX, \r\n for Windows and \r for Mac)
```
Comment (nickvergessen) [inline phpBB/includes/functions_messenger.php]: why is this changed?

File: phpBB/includes/functions_jabber.php (hunk 3 above)
Comment (MGaetan89) [inline phpBB/includes/functions_jabber.php]: `extension_loaded` returns a `bool`, so I will simply change it to `return @extension_loaded('openssl');`.

File: phpBB/includes/functions_messenger.php (hunk 4 above)
Comment (MGaetan89) [inline phpBB/includes/functions_messenger.php]: `PHP_EOL` was introduced in PHP 5.0.2. So there is no need to have a custom version.

File: phpBB/includes/functions.php (hunk 1 above)
Comment (MGaetan89) [inline phpBB/includes/functions.php]: I think I change this too quickly.
Checking if the `checkdnsrr ` function exists is enough, no? I did not find information saying if it can be disabled is some configuration of PHP.

File: phpBB/includes/functions.php (hunk 2 above)
Comment (MGaetan89) [inline phpBB/includes/functions.php]: Same as above: `if (function_exists('dns_get_record'))`.

**Comment by MGaetan89** [comment]
Once the conditions in `phpBB/includes/functions.php` are fixed, I think this PR will be ready.

File: phpBB/includes/functions.php (hunk 1 above)
Comment (nickvergessen) [inline phpBB/includes/functions.php]: I think the full if can be reduced to `if (function_exists('checkdnsrr'))`

**Comment by MGaetan89** [comment]
I updated the condition in `phpBB/includes/functions.php`. Unless you find something else to change, this PR is ready.

**Comment by MGaetan89** [comment]
!unset WIP

**Comment by MGaetan89** [comment]
Bump

**Comment by Nicofuma** [comment]
looks good to me
@nickvergessen anything else to comment?

**Comment by nickvergessen** [comment]
looks fine, but didnt look too deep into it
